pkg/tests/handlers/accounts.go

```diff
@@ -72,6 +101,81 @@ func getAuthorizationToken(connection *sdk.Connection) string {
 	return token
 }
 
+// generateClusterReRegistrationTargeter registers fake clusters and then
+// returns a targeter which uses those fake clusters.
+func generateClusterReRegistrationTargeter(qty int, url string, connection *sdk.Connection) vegeta.Targeter {
+
+	clusterIds := make([]string, qty)
+	var currentTarget = 0
+
+	// Cache the Authorization Token to avoid retrieving it with every request
+	var authorizationToken = ""
+	if len(authorizationToken) == 0 {
+		authorizationToken = getAuthorizationToken(connection)
+	}
+
+	// Register multiple mock clusters and store their IDs
+	log.Printf("Registering %d clusters to use for re-registration test", qty)
+	for i := range clusterIds {
+
+		clusterID := uuid.NewV4().String()
+
+		body, err := v1.NewClusterRegistrationRequest().AuthorizationToken(authorizationToken).ClusterID(clusterID).Build()
+		if err != nil {
+			log.Fatalln("Unable to build cluster registration request: ", err)
+		}
+
+		var rawBody bytes.Buffer
+		err = v1.MarshalClusterRegistrationRequest(body, &rawBody)
+		if err != nil {
+			log.Fatalln("Unable to serialize cluster registration request body: ", err)
+		}
+
+		resp, err := connection.AccountsMgmt().V1().ClusterRegistrations().Post().Request(body).Send()
+		if err != nil {
+			log.Fatalln("Unable to register cluster: ", err)
+		}
+
+		log.Printf("[%d/%d] Registered Cluster: '%s'. Response: %d\n", i, len(clusterIds), clusterID, resp.Status())
+		clusterIds[i] = clusterID
+
+		// Avoid hitting rate limiting
+		time.Sleep(time.Second * 1)
```

Comment:
Is this an issue?

Reply:
It is. It actually locked me out due to how fast the requests were made without sleeping, haha.